### PR TITLE
Dependent change feed implementation

### PR DESCRIPTION
**Pull request by ytqsl**
Fixes or Implements [AB#15563](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15563)

[Task 15563](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/15563): Implement #4

## Description

- added DependentAddedEvent, DependentRemovedEvent to change feed
- added integration tests
- fixed delegate unit tests due gateway api service dependent related methods refactorings

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 1]
```diff
@@ -24,31 +24,30 @@ namespace HealthGateway.Common.Messaging;
...
     /// <summary>
-    /// Initializes a new instance of the <see cref="DbOutboxStore"/> class.
+    /// Initializes a new instance of the <see cref="DbOutboxStore"/> class
     /// </summary>
-    /// <param name="dbContext">The EF database context.</param>
-    /// <param name="backgroundJobClient">Hangfire background job client.</param>
-    /// <param name="messageSender">The destination message sender to forward messages to.</param>
-    /// <param name="logger">A logger.</param>
-    public DbOutboxStore(GatewayDbContext dbContext, IBackgroundJobClient backgroundJobClient, IMessageSender messageSender, ILogger<DbOutboxStore> logger)
+    /// <param name="outboxDelegate">the outbox db delegate</param>
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:44]:
```suggestion
    /// <param name="outboxDelegate">The outbox db delegate.</param>
```
There is probably a warning regarding current param documentation. Missing capital at beginning of sentence.  Missing period at end of sentence.

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 2]
```diff
@@ -24,31 +24,30 @@ namespace HealthGateway.Common.Messaging;
...
+    /// <param name="backgroundJobClient">Hangfire background job client</param>
+    /// <param name="messageSender">The destination message sender to forward messages to</param>
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:46]:
```suggestion
    /// <param name="messageSender">The destination message sender to forward messages to.</param>
```
There is probably a warning regarding current param documentation. Missing period at end of sentence.

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 3]
```diff
@@ -24,31 +24,30 @@ namespace HealthGateway.Common.Messaging;
...
+    /// <param name="logger">A logger</param>
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:47]:
```suggestion
    /// <param name="logger">A logger.</param>
```

There is probably a warning regarding current param documentation. Missing period at end of sentence.

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 4]
```diff
@@ -69,43 +68,38 @@ public async Task StoreAsync(IEnumerable<MessageEnvelope> messages, Cancellation
...
-        await this.dbContext.SaveChangesAsync(ct);
+        await this.outboxDelegate.Queue(outboxItems, ct);
+        await this.outboxDelegate.Commit(ct);
 
         this.backgroundJobClient.Enqueue<DbOutboxStore>(store => store.DispatchOutboxItems(CancellationToken.None));
     }
 
     /// <summary>
-    /// Dispatches any outbox messages not sent.
+    /// Dispatches any outbox messages not sent
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:78]:
```suggestion
    /// Dispatches any outbox messages not sent.
```
There is probably a warning regarding current documentation. Missing period at end of sentence.

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 5]
```diff
@@ -69,43 +68,38 @@ public async Task StoreAsync(IEnumerable<MessageEnvelope> messages, Cancellation
...
     /// </summary>
-    /// <param name="ct">A cancellation token.</param>
-    /// <returns>Awaitable task.</returns>
+    /// <param name="ct">A cancellation token</param>
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:80]:
```suggestion
    /// <param name="ct">A cancellation token.</param>
```
There is probably a warning regarding current param documentation.  Missing period at end of sentence.

File: Apps/Common/src/Messaging/DbOutboxStore.cs [hunk 6]
```diff
@@ -69,43 +68,38 @@ public async Task StoreAsync(IEnumerable<MessageEnvelope> messages, Cancellation
...
+    /// <returns>Awaitable task</returns>
```
Comment (BrianMaki) [inline Apps/Common/src/Messaging/DbOutboxStore.cs:81]:
```suggestion
    /// <returns>Awaitable task.</returns>
```
There is probably a warning regarding current return documentation. Missing period at end of sentence.

**Review by BrianMaki — CHANGES_REQUESTED** [review]
Some warnings will need to be addressed/

File: Apps/Common/src/Messaging/DbOutboxStore.cs (hunk 2 above)
Comment (ytqsl) [reply Apps/Common/src/Messaging/DbOutboxStore.cs:46]: I actually disabled the warning for the . at the end, but I'll fix the rest

**Review by BrianMaki — APPROVED** [review]

File: Apps/Database/src/Delegates/DbResourceDelegateDelegate.cs [hunk 7]
```diff
@@ -87,7 +87,7 @@ public DbResult<ResourceDelegate> Insert(ResourceDelegate resourceDelegate, bool
         }
 
         /// <inheritdoc/>
-        public DbResult<IEnumerable<ResourceDelegate>> Get(string delegateId, int page, int pageSize)
+        public DbResult<IEnumerable<ResourceDelegate>> Get(string delegateId, int page = 0, int pageSize = 500)
```
Comment (BronzBierd) [inline Apps/Database/src/Delegates/DbResourceDelegateDelegate.cs:90]: Not worth rejecting but for thought:

What I've been hearing about the normal admin page size is only around 25 max. Should we be setting such a high default.

**Review by BronzBierd — APPROVED** [review]

File: Apps/Database/src/Delegates/DbResourceDelegateDelegate.cs (hunk 7 above)
Comment (ytqsl) [reply Apps/Database/src/Delegates/DbResourceDelegateDelegate.cs:90]: I fixed this in another branch

**Pull request merged**